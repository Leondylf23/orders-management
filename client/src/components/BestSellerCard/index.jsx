import { numberWithPeriods } from '@utils/allUtils';
import { FormattedMessage } from 'react-intl';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import clasess from './style.module.scss';

const BestSellerCard = ({ data }) => (
  <Link to={`/product/${data.productId}`} className={clasess.container}>
    <div className={clasess.image}>
      <img src={data.product?.imageUrl} alt="" />
    </div>
    <div className={clasess.desc}>
      <h5>{data.product?.title}</h5>
      <p className={clasess.price}>Rp. {numberWithPeriods(data.product?.price)}</p>
      <p className={clasess.org}>{data.product?.user?.organization}</p>
      <p className={clasess.sold}>
        <span>
          <FormattedMessage id="product_bought_by" />
        </span>{' '}
        {data.productCount}{' '}
        <span>
          <FormattedMessage id="times" />
        </span>
      </p>
    </div>
  </Link>
);

BestSellerCard.propTypes = {
  data: PropTypes.object,
};

export default BestSellerCard;
