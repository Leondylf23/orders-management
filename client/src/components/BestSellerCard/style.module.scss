@import '../../styles/base.scss';

.container {
  width: 350px;
  height: 120px;
  @include padding(0.5rem);
  @include directionX;
  text-decoration: none;
  color: inherit;
  border-radius: 10px;
  background-color: var(--color-bg);
  box-shadow: 5px 5px 15px rgba(0, 0, 0, 0.2);
  cursor: pointer;

  align-items: center;
  gap: 20px;
  flex-shrink: 0;

  .image {
    width: 30%;
    height: 90px;
    overflow: hidden;

    img {
      object-fit: cover;
      width: 100%;
      height: 100%;
      object-position: center;
    }
  }

  .desc {
    position: relative;
    display: flex;
    width: 70%;
    height: 100%;
    flex-direction: column;

    .price {
      font-size: 14px;
    }

    .org {
      font-size: 10px;
    }

    .sold {
      position: absolute;
      bottom: 0;
      font-size: 8px;
    }
  }
}
