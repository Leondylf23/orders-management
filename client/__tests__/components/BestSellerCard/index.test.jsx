import _ from 'lodash';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';

import store from '@store';
import Language from '@containers/Language';
import { MemoryRouter } from 'react-router-dom';

import BestSellerCard from '../../../src/components/BestSellerCard/index';
import MockBestSeller from '../../database/bestSeller.json';

let mockBestSeller;

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
}));

const ParentComponent = (children) => (
  <Provider store={store}>
    <Language>
      <MemoryRouter>{children}</MemoryRouter>
    </Language>
  </Provider>
);

describe('Best Seller Card', () => {
  beforeEach(() => {
    mockBestSeller = _.cloneDeep(MockBestSeller);
  });

  test('Correct Render', () => {
    const bestSellerCard = render(ParentComponent(<BestSellerCard data={mockBestSeller} />));
    expect(bestSellerCard.getByTestId('bestSellerCard-container')).toBeInTheDocument();
    expect(bestSellerCard.getByTestId('bestSellerCard-container')).toHaveClass('container');

    expect(bestSellerCard).toMatchSnapshot();
  });
});
