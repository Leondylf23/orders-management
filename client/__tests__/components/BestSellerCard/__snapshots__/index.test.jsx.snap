// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Best Seller Card Correct Render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="container"
        data-testid="bestSellerCard-container"
        href="/product/undefined"
      >
        <div
          class="image"
        >
          <img
            alt=""
          />
        </div>
        <div
          class="desc"
        >
          <h5 />
          <p
            class="price"
          >
            Rp. 
            0
          </p>
          <p
            class="org"
          />
          <p
            class="sold"
          >
            <span>
              Product telah dibeli
            </span>
             
             
            <span>
              kali
            </span>
          </p>
        </div>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="container"
      data-testid="bestSellerCard-container"
      href="/product/undefined"
    >
      <div
        class="image"
      >
        <img
          alt=""
        />
      </div>
      <div
        class="desc"
      >
        <h5 />
        <p
          class="price"
        >
          Rp. 
          0
        </p>
        <p
          class="org"
        />
        <p
          class="sold"
        >
          <span>
            Product telah dibeli
          </span>
           
           
          <span>
            kali
          </span>
        </p>
      </div>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
